c_src/TLH_absTab.c

```diff
@@ -0,0 +1,225 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <glib.h>
+#include "absTab.h"
+#include "message.h"
+
+struct AbsTab {
+    /* a two-level hash table where the value of each 
+     * key points to another (level2) hash table 
+     */
+    GHashTable *lev1; /* directory watch descriptors */
+    GHashTableIter lev1_rmItr, lev2_rmItr; /* used for remove_next_absTab() */
+    int rmItr_initialized;
+    int *lev1_rmItr_key;
+    GHashTable *lev1_rmItr_val;
+    /* used for remove_next_absTab(). This is not very nice,
+     * but I need lev1_rmItr_key (directory wd) because glib
+     * does not provide any API for peeking keys.
+     */
+};
+
+AbsTab_t new_absTab()
```

Comment:
Agreed. Kind of noticed this myself towards the end of my coding!
Will update accordingly.